internal/command/action.go

```diff
@@ -14,40 +16,42 @@ import (
 )
 
 type ActionConfig struct {
-	Args        []string
-	Provider    string
-	Filepath    string
-	YamlInline  string
-	Subs        map[string]string
-	Ignores     []string
-	IgnoreAll   bool
-	Environment string
+	Args           []string
+	Provider        string
+	Filepath        string
+	YamlInline      string
+	Subs            map[string]string
+	Ignores         []string
+	IgnoreAll       bool
+	Environment     string
+	AllProvVersions bool
 }
 
 const ENV_FILE_MAGIC = "@SUMMONENVFILE"
 const SUMMON_ENV_KEY_NAME = "SUMMON_ENV"
 
 var Action = func(c *cli.Context) {
-	if !c.Args().Present() {
+	if !c.Args().Present() && !c.Bool("all-provider-versions") {
 		fmt.Println("Enter a subprocess to run!")
 		os.Exit(127)
 	}
 
 	provider, err := prov.Resolve(c.String("provider"))
-	if err != nil {
+	if err != nil && !c.Bool("all-provider-versions") {
```

Comment:
It would be good to leave a note here why we are ok with provider not loading here if that flag is set